src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java

```diff
@@ -224,34 +230,38 @@ public void generateTerrain2(int chunkX, int chunkZ, ChunkPrimer primer) {
 	}
 
 	private void makeLandPerBiome2(int x, int zero, int z) {
-
-		this.field_147426_g = this.noiseGen6.generateNoiseOctaves(this.field_147426_g, x, z, 5, 5, 200.0D, 200.0D, 0.5D);
-		this.field_147427_d = this.field_147429_l.generateNoiseOctaves(this.field_147427_d, x, zero, z, 5, 33, 5, 8.555150000000001D, 4.277575000000001D, 8.555150000000001D);
-		this.field_147428_e = this.field_147431_j.generateNoiseOctaves(this.field_147428_e, x, zero, z, 5, 33, 5, 684.412D, 684.412D, 684.412D);
-		this.field_147425_f = this.field_147432_k.generateNoiseOctaves(this.field_147425_f, x, zero, z, 5, 33, 5, 684.412D, 684.412D, 684.412D);
+		this.noiseGen6Values = this.noiseGen6.generateNoiseOctaves(this.noiseGen6Values, x, z, 5, 5, 200.0D, 200.0D, 0.5D);
+		this.octaves3Values = this.octaves3.generateNoiseOctaves(this.octaves3Values, x, zero, z, 5, 33, 5, 8.555150000000001D, 4.277575000000001D, 8.555150000000001D);
+		this.octaves1Values = this.octaves1.generateNoiseOctaves(this.octaves1Values, x, zero, z, 5, 33, 5, 684.412D, 684.412D, 684.412D);
+		this.octaves2Values = this.octaves2.generateNoiseOctaves(this.octaves2Values, x, zero, z, 5, 33, 5, 684.412D, 684.412D, 684.412D);
 		int terrainIndex = 0;
 		int noiseIndex = 0;
-
+		
 		for (int ax = 0; ax < 5; ++ax) {
-			for (int az = 0; az < 5; ++az) {
+			int biomeForGenIndex = ax + 2 + 2 * 10;
+			
+			for (int az = 0; az < 5; ++az, biomeForGenIndex += 10) {
 				float totalVariation = 0.0F;
 				float totalHeight = 0.0F;
 				float totalFactor = 0.0F;
 				byte two = 2;
-				Biome biomegenbase = this.biomesForGeneration[ax + 2 + (az + 2) * 10];
+				Biome biomegenbase = this.biomesForGeneration[biomeForGenIndex/*ax + 2 + (az + 2) * 10*/];
```

Comment:
don't leave random things commented in, just remove it if you're confident in the change